src/Components/SmartComponents/SystemsPage/SystemsPage.js

```diff
@@ -90,15 +93,17 @@ const SystemsPage = ({ intl }) => {
     };
 
     useEffect(() => {
-        if (!inventory.current) {
-            apply(urlParameters);
-        }
-        else {
-            dispatch(fetchSystems(parameters));
+        apply(urlParameters);
+        setFirstMount(false);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I wish we could do something about this annoying rule.

Reply:
@leSamo, At first,  I feel it's bad disabling the rule!!! (**even here that I put it**).
React provides the eslint rule to follow good practice.  

You have a great point anyway, I think we need to start  using/understand `useEffect` better. 
Few articles that might help up 

1. https://reactjs.org/docs/hooks-faq.html#performance-optimizations
2. https://overreacted.io/a-complete-guide-to-useeffect/